loris/resolver.py

```diff
@@ -177,20 +178,10 @@ def is_resolvable(self, ident):
         else:
             fp = self._web_request_url(ident)
 
-            if self.head_resolvable:
```

Comment:
I really like the `with closing(` statement, but I'm pretty sure that other users (@scande3 @rlskoeser ?) need that head request check. If you can update lines 181 - 193 to use a similar syntax, I think that should be the plan.

No need for a new PR unless you want to--I'm fine with you just updating this branch and squashing the commits.